### PR TITLE
convert some missing class.

### LINKED ISSUE #3758: ES6 Transition: Concerns & Progress
Hello everyone! 

I'm investigating the means of fully transitioning p5 to ES6. I've already spoken to some of the developers outside of Github and was hoping to shine some light on some of the intricacies that I've observed in the code.

I've noticed there are 42 instances of `module.exports = p5;` in the bundled .js files. And I hope this issue serves as a clarification of the assumption that I have:

From what I understand each `module.exports = p5;` is effectively exposing p5 after extending it in its own respected files/module, be it curves.js, vertex.js, etc. 

Although what seems to be the case is that most modules usually inherit/require from main.js, there are few instances of sequential inheritance of p5. 

For instance:
In `string_functions.js` we have:
```
var p5 = require('../core/main');
require('../core/error_helpers');
....
module.exports = p5;
```
Both `main` and `error_helpers` have their own respective `modules.exports = p5`. However `string_functions` is exporting the instance of p5 inheritted from `main`.

My question in regards to this specific instance is by the end of `string_functions` when we are exporting p5, will this instance have all the prototypal functions defined in `error_helpers`? 

From my understanding, the inheritance scheme seems to be a directed acyclical graph with a single root node (`main.js` - **A** in picture) and a single leaf node which is p5 at its final stage after all the additions to the prototype occur in multiple files ( **F** in picture or equivalently `p5()` in browser).

1sQcP

I know javascript is independent of the order the code is written unlike C++, and instances and references are resolved during JIT. Given that there are 42 instances of `module.exports=p5` in the bundled lib, how is each instance resolved properly? Is each instance unique, with functionalities unavailable in preceding exports or do they all refer to the same class for lack of a better word?

Moreover given the `import` syntax in ES6, how does this export/require paradigm change, if it does?

Apologies in advance if these questions seem trivial or show a lack of knowledge in JS. I come from different programming languages and JS seems like wild west to me!

Edit: I'm unable to appropriately label my issue. I assume it should be considered a "discussion".

### PR DESCRIPTION
**Pull request by wuyudi**
Resolves #3758

More class to es6, this is a continue to https://github.com/processing/p5.js/pull/6075

 Changes:

I use `p5\..*?.prototype\..*?` to find them.

 Screenshots of the change:

#### PR Checklist

- [x] `npm run lint` passes
- [x] [Inline documentation] is included / updated
- [ ] [Unit tests] are included / updated

[Inline documentation]: https://github.com/processing/p5.js/blob/main/contributor_docs/inline_documentation.md
[Unit tests]: https://github.com/processing/p5.js/tree/main/contributor_docs#unit-tests

### CODE REVIEW

File: src/webgl/3d_primitives.js [hunk 1]
```diff
@@ -174,8 +174,8 @@ p5.prototype.box = function(width, height, depth, detailX, detailY) {
         [20, 21],
         [22, 23]
       ];
-      for (let i = 0; i < cubeIndices.length; i++) {
-        const cubeIndex = cubeIndices[i];
+
+      cubeIndices.forEach((cubeIndex, i) => {
```
Comment (limzykenneth) [inline src/webgl/3d_primitives.js:178]: Not sure if it is still true or relevant but `forEach` tends to be slower than pure for loops.

Comment (wuyudi) [reply src/webgl/3d_primitives.js:178]: https://measurethat.net/Benchmarks/Show/2090/0/array-loop-vs-foreach-vs-map#latest_results_block it seems that forEach now is faster.

Comment (limzykenneth) [reply src/webgl/3d_primitives.js:178]: Just tried that benchmark on my computer and for Chrome foreach is faster but for Firefox for loops are about 5-6x faster. A bit surprised Chrome is optimized for foreach though. 🤔
Screenshot 2023-06-01 at 16 11 30

**Comment by wuyudi** [comment]
[Why there is a remaining `p5.RendererGL.prototype.tessyVertexSize = 12;`](https://stackoverflow.com/questions/34185643/ecmascript-6-syntax-for-declaring-prototype-default-values-using-class)

**Comment by wuyudi** [comment]
I think now it's ok to merge :eyes:

**Comment by jeffscottward** [comment]
I'm late to the conversion party but just wondering why use a classes?

Definitely agree we have to get away from prototype but at least in the react.js (react team moved away after using it) and node.js worlds, classes are frowned upon. It's a jarring context switch for a language that is designed to be functional.

Is using classes required for this switch away from prototype?

**Comment by davepagurek** [comment]
> Is using classes required for this switch away from prototype?

I would say that it's not required, but is the easiest change, since the previous code was already using classes but just using an ES5 syntax for it, so we could change to `class` syntax without updating any call sites.

**Comment by jeffscottward** [comment]
> > Is using classes required for this switch away from prototype?
> 
> 
> 
> I would say that it's not required, but is the easiest change, since the previous code was already using classes but just using an ES5 syntax for it, so we could change to `class` syntax without updating any call sites.

Oh that is smart actually. Ok fair. :) I might try to throw AI at this and see if it can blaze through.

**Comment by jeffscottward** [comment]
> > > Is using classes required for this switch away from prototype?
> > 
> > 
> > I would say that it's not required, but is the easiest change, since the previous code was already using classes but just using an ES5 syntax for it, so we could change to `class` syntax without updating any call sites.
> 
> Oh that is smart actually. Ok fair. :) I might try to throw AI at this and see if it can blaze through.

Uh oh..
https://github.com/processing/p5.js/blob/main/contributor_docs/archive/es6-adoption.md#browser-compatibility-and-transpiling

http://incaseofstairs.com/six-speed/

Assuming we are using Babel - Classes is a *massive* perf hit. 8x? wow.
What about typescript?

Also - based on Contrib docs - seems like a master Issue for ES6 conversion is missing? These one of PR's leave the conversation sporadic in closed issue comments. Should I make one?

**Comment by jeffscottward** [comment]
🚨 🚨  Please move ES6 conversion commentary over to https://github.com/processing/p5.js/issues/6371 🚨 🚨

**Pull request merged**